Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -0,0 +1,62 @@
+{ Expression } = require './expression'
+{ FunctionRef } = require './reusable'
+{ ValueSet, Code } = require '../datatypes/datatypes'
+{ build } = require './builder'
+
+module.exports.Quantity = class Quantity extends Expression
+  constructor: (json) ->
+    super
+    @unit = json.unit
+    @value = json.value
+
+  exec: (ctx) ->
+    @  
+  sameOrBefore: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value <= other.value else null
+
+  sameOrAfter: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value >= other.value else null
+
+  after: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value > other.value else null
+
+  before: (other) ->
+    if other instanceof Quantity and other.unit == @unit then @value < other.value else null
+
+time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+
+clean_unit = (units) ->
+  if time_units[units] then time_units[units] else units
+
+module.exports.createQuantity = (value,unit) ->
+  new Quantity({value: value, unit: unit})
+
+module.exports.doAddition = (a,b) ->
+  if a instanceof Quantity and b instanceof Quantity
+    if a.unit == b.unit 
+      new Quantity({unit: a.unit, value: a.value + b.value})
+  else
+    a.copy().add(b.value, clean_unit(b.unit))
+
+
+module.exports.doSubtraction = (a,b) ->
+  if a instanceof Quantity and b instanceof Quantity
+    if a.unit == b.unit 
+      new Quantity({unit: a.unit, value: a.value - b.value})
+  else
+    a.copy().add(b.value * -1 , clean_unit(b.unit))  
+
+module.exports.doDivision = (a,b) ->
+  if a instanceof Quantity and b instanceof Quantity
+    if a.unit == b.unit 
+      new Quantity({unit: a.unit, value: a.value / b.value})
+  else
+    new Quantity({unit: a.unit, value: a.value / b})
+
+module.exports.doMultiplication = (a,b) ->
+  if a instanceof Quantity and b instanceof Quantity
+    if a.unit == b.unit 
+      new Quantity({unit: a.unit, value: a.value * b.value})
```

Comment:
I think this is wrong.  5m * 5m is 25m<sup>2</sup> (_not_ 25m).  Figuring out the proper square unit notation is probably not something we want to do right now, so we probably should just throw an error or return null or something.